pkg/pattern/instance.go

```diff
@@ -78,10 +79,14 @@ func (i *instance) Iterator(ctx context.Context, req *logproto.QueryPatternsRequ
 		return nil, httpgrpc.Errorf(http.StatusBadRequest, err.Error())
 	}
 	from, through := util.RoundToMilliseconds(req.Start, req.End)
+	step := req.Step
+	if step < 10*time.Second.Milliseconds() {
```

Comment:
Should this be more `step < timeResolution` ? Easy to miss in the future.

Reply:
Good idea. I exposed drain.TimeResolution to do this. IMO it makes sense to do this at the API layer than to default it later in the call stack where it's harder to surface what happened and why. I don't feel strongly about this though.